src/Sylius/Bundle/ResourceBundle/spec/Controller/ResourcesCollectionProviderSpec.php

```diff
@@ -164,4 +164,66 @@ function it_creates_a_pagniated_respresentation_for_pagerfanta_for_non_html_requ
 
         $this->get($requestConfiguration, $repository)->shouldReturn($paginatedRepresentation);
     }
+
+    function it_sets_current_page_on_paginator_from_custom_method(
+        RequestConfiguration $requestConfiguration,
+        RepositoryInterface $repository,
+        Pagerfanta $paginator,
+        Request $request,
+        ParameterBag $queryParameters
+    ) {
+        $requestConfiguration->isHtmlRequest()->willReturn(true);
+        $requestConfiguration->getRepositoryMethod()->willReturn('findAll');
+        $requestConfiguration->getRepositoryArguments()->willReturn(['foo']);
+
+        $requestConfiguration->isPaginated()->willReturn(false);
+        $requestConfiguration->isLimited()->willReturn(true);
+        $requestConfiguration->getLimit()->willReturn(15);
+
+        $repository->findAll('foo')->willReturn($paginator);
+
+        $requestConfiguration->getRequest()->willReturn($request);
+        $request->query = $queryParameters;
+        $queryParameters->get('page', 1)->willReturn(8);
+
+        $paginator->setCurrentPage(8)->shouldBeCalled();
+
+        $this->get($requestConfiguration, $repository)->shouldReturn($paginator);
+    }
+
+    function it_creates_a_pagniated_respresentation_for_pagerfanta_for_non_html_requests_with_a_custom_repository_method(
+        RequestConfiguration $requestConfiguration,
+        RepositoryInterface $repository,
+        Pagerfanta $paginator,
+        Request $request,
+        ParameterBag $queryParameters,
+        ParameterBag $requestAttributes,
+        PagerfantaFactory $pagerfantaRepresentationFactory,
+        PaginatedRepresentation $paginatedRepresentation
+    ) {
+        $requestConfiguration->isHtmlRequest()->willReturn(false);
+        $requestConfiguration->getRepositoryMethod()->willReturn('findAll');
+        $requestConfiguration->getRepositoryArguments()->willReturn(['foo']);
+
+        $requestConfiguration->isPaginated()->willReturn(true);
+        $requestConfiguration->isLimited()->willReturn(false);
+        $requestConfiguration->getCriteria()->willReturn([]);
+        $requestConfiguration->getSorting()->willReturn([]);
+
+        $repository->findAll('foo')->willReturn($paginator);
```

Comment:
The point was to demonstrate that it works with a custom method, other than ``createPaginator``. If I change it to ``createPaginator`` that spec would pass even if custom method is not used at all/not working, etc. Ideally, we can introduce "TestRepositoryInterface" just for this spec with something really custom. I'd do that later though and get this fixed now.